packages/tupaia-web/src/features/Map/MapOverlaySelector/MapTableModal.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import styled from 'styled-components';
+import DownloadIcon from '@material-ui/icons/GetApp';
+import MuiIconButton from '@material-ui/core/IconButton';
+import { FlexColumn } from '@tupaia/ui-components';
+import { Modal } from '../../../components';
+import { MapTable, MeasureData } from '@tupaia/ui-map-components';
+import { useParams } from 'react-router';
+import { useEntityAncestors, useMapOverlays, useEntitiesWithLocation } from '../../../api/queries';
+import { useMapOverlayReport } from '../utils';
+import { processMeasureData } from '../MapOverlays/processMeasureData';
+
+const Wrapper = styled(FlexColumn)`
+  justify-content: flex-start;
+`;
+
+const Title = styled.div.attrs({
+  variant: 'h2',
+})`
+  font-size: 1.25rem;
+  line-height: 1.4;
+  font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  margin-bottom: 1.5rem;
+`;
+
+const IconButton = styled(MuiIconButton)`
+  margin-bottom: 1.5rem;
+`;
+
+const TitleWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  margin-bottom: 3rem;
+  position: relative;
+  border-bottom: 1px solid rgba(255, 255, 255, 0.12);
+`;
+
+const useEntitiesByMeasureLevel = (measureLevel?: string) => {
```

Comment:
I'm guessing this is copied over from `DataVisualsLayer`? I think instead of copying it completely, we could move it to the `utils` folder, and import it wherever we use it, ie. in these two files.